protocol/tags.go

```diff
@@ -46,6 +46,90 @@ const (
 	VoteBundleTag Tag = "VB"
 )
 
+// The following constants are overestimates in some cases but are reasonable upper bounds
+// for the purposes of limiting the number of bytes read from the network.
+// The calculations to obtain them are defined in node/TestMaxSizesCorrect()
+
+// AgreementVoteTagMaxSize is the maximum size of an AgreementVoteTag message
+const AgreementVoteTagMaxSize = 1228
+
+// MsgOfInterestTagMaxSize is the maximum size of a MsgOfInterestTag message
+const MsgOfInterestTagMaxSize = 45
+
+// MsgDigestSkipTagMaxSize is the maximum size of a MsgDigestSkipTag message
+const MsgDigestSkipTagMaxSize = 69
+
+// NetPrioResponseTagMaxSize is the maximum size of a NetPrioResponseTag message
+const NetPrioResponseTagMaxSize = 838
+
+// NetIDVerificationTagMaxSize is the maximum size of a NetIDVerificationTag message
+const NetIDVerificationTagMaxSize = 215
+
+// PingTagMaxSize is the maximum size of a PingTag message
+const PingTagMaxSize = 8
+
+// PingReplyTagMaxSize is the maximum size of a PingReplyTag message
+const PingReplyTagMaxSize = 8
+
+// ProposalPayloadTagMaxSize is the maximum size of a ProposalPayloadTag message
+// This value is dominated by the MaxTxnBytesPerBlock
+const ProposalPayloadTagMaxSize = 5247980
+
+// StateProofSigTagMaxSize is the maximum size of a StateProofSigTag message
+const StateProofSigTagMaxSize = 6378
+
+// TopicMsgRespTagMaxSize is the maximum size of a TopicMsgRespTag message
+// This is a response to a topic message request (either UE or MI) and the largest possible
+// response is the largest possible block.
+// Matches  current network.MaxMessageLength
+const TopicMsgRespTagMaxSize = 6 * 1024 * 1024
+
+// TxnTagMaxSize is the maximum size of a TxnTag message. This is equal to SignedTxnMaxSize()
+// which is size of just a single message containing maximum Stateproof. Since Stateproof
+// transactions can't be batched we don't need to multiply by MaxTxnBatchSize.
+const TxnTagMaxSize = 4620031
+
+// UniEnsBlockReqTagMaxSize is the maximum size of a UniEnsBlockReqTag message
+const UniEnsBlockReqTagMaxSize = 67
+
+// VoteBundleTagMaxSize is the maximum size of a VoteBundleTag message
+// Matches current network.MaxMessageLength
+const VoteBundleTagMaxSize = 6 * 1024 * 1024
+
+// MaxMessageSize returns the maximum size of a message for a given tag
+func (tag Tag) MaxMessageSize() uint64 {
```

Comment:
Instead of a manually maintained constants with switch statement, why not a map?

```
MaxMessageSizes := map[tag]Value {
VoteBundle: 6 * 1024 * 1024,
...
}
...

func MaxMessageSize() {
return MaxMessageSizes[tag.Name]
}
```
(awful pseudocode, but you get the idea)

Reply:
this switch appears ~15% faster